ocs_ci/ocs/resources/mcg.py

```diff
@@ -73,20 +77,33 @@ def __init__(self):
             }
         ).json().get('reply').get('token')
 
+        (
+            self.cred_req_obj,
+            self.aws_access_key_id,
+            self.aws_access_key
+        ) = self.request_aws_credentials()
+
         self._ocp_resource = ocp_obj
+
         self.s3_resource = boto3.resource(
             's3', verify=False, endpoint_url=self.s3_endpoint,
             aws_access_key_id=self.access_key_id,
             aws_secret_access_key=self.access_key
         )
 
+        self.aws_s3_resource = boto3.resource(
+            's3', verify=False, endpoint_url="https://s3.amazonaws.com",
+            aws_access_key_id=self.aws_access_key_id,
+            aws_secret_access_key=self.aws_access_key
+        )
+
     def s3_list_all_bucket_names(self):
         """
         Returns:
             list: A list of all bucket names
 
         """
-        return [bucket.name for bucket in self.s3_resource.buckets.all()]
+        return {bucket.name for bucket in self.s3_resource.buckets.all()}
```

Comment:
Forgot to refactor the docs and names - that's a mistake on my side.
I wanted to update it to a set to keep the performance as good as possible. Bucket names are, by definition, unique across the entire service. Changing this from a list to a set allows us to have an access efficiency of O(1) instead of O(n).

Reply:
I didn't find the updated functions been used with list get, so it did not break anything, so update to set is fine.